### PR TITLE
refactor: use ops for idna & punycode

### PR DESCRIPTION
**Pull request by crowlKats**
Towards https://github.com/denoland/deno/issues/17809

Unsure if we want to get rid of ucs2 namespace or not; doesnt seem very used. if we get rid of it, we can get rid of the entire idna.ts file.

### CODE REVIEW

File: ext/node/idna.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+) -> Result<String, AnyError> {
+  Ok(idna::domain_to_ascii(&domain)?)
+}
+
+#[op]
+pub fn op_node_idna_domain_to_unicode(
+  domain: String,
+) -> Result<String, AnyError> {
+  Ok(idna::domain_to_unicode(&domain).0)
+}
```
Comment (littledivy) [inline ext/node/idna.rs:18]: Just `String` instead of `Result<String, E>`

File: ext/node/idna.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+#[op]
+pub fn op_node_idna_punycode_decode(
+  domain: String,
+) -> Result<String, AnyError> {
```
Comment (littledivy) [inline ext/node/idna.rs:23]: Ditto

File: ext/node/idna.rs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  Ok(idna::punycode::decode_to_string(&domain).unwrap_or_default())
+}
+
+#[op]
+pub fn op_node_idna_punycode_encode(
+  domain: String,
+) -> Result<String, AnyError> {
```
Comment (littledivy) [inline ext/node/idna.rs:30]: Ditto

**Review by bartlomieju — APPROVED** [review]
LGTM

**Review by littledivy — APPROVED** [review]
LGTM 2

**Pull request merged**